plugin/trino-loki/src/main/java/io/trino/plugin/loki/LokiMetadata.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.loki;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.github.jeschkies.loki.client.LokiClient;
+import io.github.jeschkies.loki.client.LokiClientException;
+import io.github.jeschkies.loki.client.model.Data;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorMetadata;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.TableFunctionApplicationResult;
+import io.trino.spi.function.table.ConnectorTableFunctionHandle;
+import io.trino.spi.type.DoubleType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeManager;
+import io.trino.spi.type.VarcharType;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.loki.LokiTableFunction.SCHEMA_NAME;
+import static io.trino.spi.type.TimestampWithTimeZoneType.createTimestampWithTimeZoneType;
+import static io.trino.spi.type.TypeSignature.mapType;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class LokiMetadata
+        implements ConnectorMetadata
+{
+    private static final String TABLE_NAME = "default";
```

Comment:
What would happen after running a `SELECT * FROM default` query? Can Trino throw an "unsupported" exception, or will this actually try executing some query in Loki?

Reply:
EDIT: Great feedback. I've added a custom error.